packages/frontend/src/components/common/modal/ChartUpdateModal.tsx

```diff
@@ -24,8 +25,12 @@ const ChartUpdateModal: FC<ChartUpdateModalProps> = ({
     onConfirm,
     ...modalProps
 }) => {
+    const dashboardUuid = useSearchParams('fromDashboard');
     const { data: chart, isLoading } = useSavedQuery({ id: uuid });
-    const { mutateAsync, isLoading: isUpdating } = useUpdateMutation(uuid);
+    const { mutateAsync, isLoading: isUpdating } = useUpdateMutation(
+        dashboardUuid!,
```

Comment:
we cannot enforce a dashboardUuid here, this works if you are updating a chart from a dashboard, but you can also update charts without a dashboard . This field needs to be optional (string | undefined) and only redirect if provided


[Screencast from 31-03-23 16:43:39.webm](https://user-images.githubusercontent.com/1983672/232403955-4e1a0cb0-5512-4796-af25-778de690357b.webm)